zproject/backends.py

```diff
@@ -833,7 +840,16 @@ def sync_user_from_ldap(user_profile: UserProfile, logger: logging.Logger) -> bo
     try:
         ldap_username = backend.django_to_ldap_username(user_profile.delivery_email)
     except ZulipLDAPExceptionNoMatchingLDAPUser:
-        if (
+        # Check if the mail has to be updated or the user has to be disabled
+
+        if user_profile.ext_auth_uid is not None and 'ldap' in user_profile.ext_auth_uid:  # type: ignore[operator] # Somehow mypy thinks in is not acceptable here # nocoverage
+            pass
+            # TODO
+            # user = settings.AUTH_LDAP_USER_SEARCH.execute ???
+            # if user is not None:
+            #    mail = backend.user_email_from_ldapuser(user_profile.ext_auth_uid['ldap'], user)
+            #    do_change_user_delivery_email(user_profile, mail)
```

Comment:
user_profile <-> ldap syncing happen in `ZulipLDAPUserPopulator.get_or_build_user`, so that's where the logic should go